e2e/specs/translation-builder-workflows.spec.ts

```diff
@@ -0,0 +1,178 @@
+      await expect(results.first()).toBeVisible();
+    });
+  });
+
+  test('Edit and update an individual translation', async ({ page }) => {
+    const formBuilderPage = new FormBuilderPage(page);
+
+    await test.step('When I open a form in the Translation Builder', async () => {
+      await formBuilderPage.gotoFormBuilder();
+      await formBuilderPage.searchForForm(formDetails.name);
+      await formBuilderPage.page.getByRole('row', { name: formDetails.name }).getByLabel('Edit Schema').first().click();
+    });
+
+    await test.step('And then I go to the translation builder', async () => {
+      await formBuilderPage.translationBuilderTab().click();
+    });
+
+    await test.step('And then I select the French language', async () => {
+      await formBuilderPage.languageDropdown().click();
+      const dropdownMenu = formBuilderPage.translationBuilderPanel().locator('.cds--list-box__menu');
+      await expect(dropdownMenu).toBeVisible();
+      await formBuilderPage.page.getByRole('option', { name: 'French (fr)' }).click();
+    });
+
+    await test.step('And then I edit a translation string and save it', async () => {
+      await formBuilderPage.editTranslationButton(0).click();
+      const translationInput = formBuilderPage.translationValueInput();
+      await translationInput.fill('Test Translation Updated');
+    });
+    await test.step('And then when I preview the form in the language, I should see the edited translation strings', async () => {
+      await formBuilderPage.saveTranslationButton().click();
+      await expect(formBuilderPage.translationModal()).toBeHidden();
+      await expect(formBuilderPage.editTranslationButton(0)).toBeEnabled();
+    });
+  });
+  test('Download a translation file from a form', async ({ page }) => {
+    const formBuilderPage = new FormBuilderPage(page);
+
+    await test.step('When I open a form in the translation Builder', async () => {
+      await formBuilderPage.gotoFormBuilder();
+      await formBuilderPage.searchForForm(formDetails.name);
+      await formBuilderPage.page.getByRole('row', { name: formDetails.name }).getByLabel('Edit Schema').first().click();
+    });
+
+    await test.step('And then I go to the translation builder', async () => {
+      await formBuilderPage.translationBuilderTab().click();
+    });
+
+    await test.step('And I download the translation file', async () => {
+      const downloadButton = formBuilderPage.downloadTranslationButton();
+      const [download] = await Promise.all([
+        formBuilderPage.page.waitForEvent('download', { timeout: 15000 }),
+        downloadButton.click(),
+      ]);
+      expect(download.suggestedFilename()).toMatch(/\.json$/);
+    });
+  });
+
+  test('Upload a translation file to backend', async ({ page }) => {
+    const formBuilderPage = new FormBuilderPage(page);
+
+    await test.step('When I open a form in the translation builder', async () => {
```

Comment:
Let's change the step names,

1. Given I have opened a form in the Form Builder
2. When I navigate to the Translation Builder
3. Then I should be able to upload a translation file successfully